File: B_PL_Codes/15-Ponteiros_basico/pont03.c
/*
Lógica de programação
Prof. Jonatha Costa
Exercício resolvido com ponteiros.
Escreva um código em C que preencha um vetor com 10 números e retorne quais são os números ímpares deste vetor.
*/

#include <stdio.h>

int main() {
    int v[10], i, *p;

    printf("Digite 10 números:\n");
    for (p = v; p < v + 10; p++) {
        scanf("%d", p);
    }

    printf("Números ímpares:\n");
    for (p = v; p < v + 10; p++) {
        if (*p % 2 != 0)
            printf("%d ", *p);
    }

    printf("\n");
    return 0;
}

File: B_PL_Codes/15-Ponteiros_basico/InJava/pont03.java
/*
Lógica de programação
Prof. Jonatha Costa
Exercício resolvido em java.
Escreva um código em C que preencha um vetor com 10 números e retorne quais são os números ímpares deste vetor.
*/

import java.util.Scanner;

public class pont03 {

    public static void main(String[] args) {
        Scanner dados = new Scanner(System.in);

        int[] v = new int[10];
        int i;
        System.out.printf("Digite 10 números:\n");
        for (i = 0; i < 10; i++) {
            v[i] = dados.nextInt();
        }

        System.out.printf("Números ímpares:\n");
        for (i = 0; i < 10; i++) {
            if (v[i] % 2 != 0)
                System.out.printf("%d ", v[i]);
        }

        System.out.printf("\n");
        dados.close();
    }
}
